### ISSUE TITLE
Allow using helper methods outside content blocks

### ISSUE THREAD

**Issue by TastyPi** [jonspalmer/view_component-storybook#128]
I'm building a component that delegates to [`select_tag`](https://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html#method-i-select_tag) to render a slot. The slot directly passes the `option_tags` parameter straight through. It would be nice if I could call `options_for_select` to create the `option_tags` parameter value, but helper methods are not available at the point where slot parameters are set.

Currently I'm working around this by just manually creating the string, but it would be nice if I could call helpers.

What I want:

```ruby
story(:my_select) do
  my_select_slot(options_for_select(["1", "2", "3"]))
end
```

What I'm doing:

```ruby
story(:my_select) do
  my_select_slot("<option value=\"1\">1</option><option value=\"2\">2</option><option value=\"3\">3</option>".html_safe)
end
```

**Comment by jonspalmer** [comment]
This is an interesting idea. I think the challenge is at story definition time we don't have a view context from which to leverage the helpers such as options_for_select. Perhaps they're available statically and for helpers that don't rely on the request explicitly they work. Have you tried simply including the helper classes in the Stories class?

**Comment by TastyPi** [comment]
We ended up moving away from this library, instead building our own Storybook infrastructure  using templates. We felt this library was fairly restrictive (e.g. lack of polymorphic slot support, lack of a view context for helper methods), and it felt like having to learn a new language for rendering slots - using templates means rendering a component is done in exactly the same inside and outside a story.

This was a great starting point for us to experiment with using Storybook and decide we want to commit to it, but since we are no longer using it I unfortunately don't have time to investigate ways I could have used the helper method. I look forward to coming back to this project and thinking it looks way better than what we built.

**Comment by jonspalmer** [comment]
@TastyPi I'm curious about the template approach. Are you still using Storybook/server to render the stories in Javascript etc. Just a different backend for the html content?

**Comment by TastyPi** [comment]
Yeah, the basic idea is similar to this gem, there's some code to generate the JSON files and a Controller that maps paths to view templates.

However, we're probably going to switch to https://github.com/ViewComponent/lookbook, which is a Storybook-like interface but designed to work with ViewComponents. Haven't had time to fully investigate it though.

**Comment by jonspalmer** [comment]
Circling back to this conversation I'm curious if you have thoughts on the discussion in #150

**Comment by jonspalmer** [comment]
fixed in v1.0.0. 

Stories are now just extensions of ViewComponent Previews and support all options they support
